### ISSUE TITLE
Handle unsupported pixel types better to prevent SIGABRT

### ISSUE THREAD

**Issue by rheniser** [Sosoyan/Aton#50]
AtoN appears to only support a subset of Arnold pixel types (FLOAT, RGB, RGBA, VECTOR, and POINT). I did not find this documented anywhere outside of the code. When one uses one of the unsupported pixel types, for example INT, the Boost::Asio library throws an opaque exception (to most programmers, except for the few well-versed in the Boost::Asio library):

~~~~
Backtrace from AtoN Arnold plug-in (driver_aton.so):


terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::system::system_error> >'
  what():  write: Bad address
00:00:09   292MB ERROR   |   signal caught: SIGABRT -- Abnormal process abort

****
* Arnold 4.2.16.0 linux clang-3.8.0 glibc-2.12 oiio-1.7.7 rlm-12.0.2 2016/11/22 13:32:54
00:00:09   292MB         |     5% done - 26 rays/pixel
* CRASHED in gsignal at 00:00:00
* signal caught: SIGABRT -- Abnormal process abort
*
* backtrace:
*  0 0x00007f6b96e37790 [libai.so       ]                                                                                                                                                                                                                                                                                                                                                
*  1 0x000000377723269f [libc.so.6      ]                                                                                                                                                                                                                                                                                                                                                
>> 2 0x0000003777232624 [libc.so.6      ] gsignal                                                                                                                                                                                                                                                                                                                                        
*  3 0x0000003777232624 [libc.so.6      ] gsignal                                                                                                                                                                                                                                                                                                                                        
*  4 0x00007f6b966c3a6e [libstdc++.so.6 ] __gnu_cxx::__verbose_terminate_handler()                                                                                                                                                                                                                                                                                                       
*  5 0x00007f6b966c0c61 [libstdc++.so.6 ]                                                                                                                                                                                                                                                                                                                                                [eh_unex_handler.cc:0]
*  6 0x00007f6b966c0c89 [libstdc++.so.6 ] __cxxabiv1::__terminate(void (*)                                                                                                                                                                                                                                                                                                               
*  7 0x00007f6b966c10f3 [libstdc++.so.6 ] __cxa_rethrow                                                                                                                                                                                                                                                                                                                                  
*  8 0x00007f6b4c9455ce [driver_aton.so ] void boost::throw_exception<boost::system::system_error>(boost::system::system_error const&)                                                                                                                                                                                                                                                   
*  9 0x00007f6b4c9456f7 [driver_aton.so ] boost::asio::detail::do_throw_error(boost::system::error_code const&, char const*)                                                                                                                                                                                                                                                             
* 10 0x00007f6b4c946b70 [driver_aton.so ] boost::asio::detail::do_throw_error(boost::system::error_code const&, char const*)                                                                                                                                                                                                                                                             
* 11 0x00007f6b4c940ebb [driver_aton.so ] Client::sendPixels(Data&)                                                                                                                                                                                                                                                                                                                      
* 12 0x00007f6b4c93f95a [driver_aton.so ] unsigned long boost::asio::write<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, boost::asio::mutable_buffers_1>(boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >&, boost::asio::mutable_buffers_1 const&) [Driver_Aton.cpp   :0]
* 13 0x00007f6b96e06969 [libai.so       ]                                                                                                                                                                                                                                                                                                                                                
* 14 0x00007f6b96e3a909 [libai.so       ]                                                                                                                                                                                                                                                                                                                                                
* 15 0x00000037776079d0 [libpthread.so.0]                                                                                                                                                                                                                                                                                                                                                [pthread_create.c  :0]
* 16 0x00000037772e88fc [libc.so.6      ] clone                                                                                                                                                                                                                                                                                                                                          
*
* loaded modules:
*    0x00007f6b96bb5000  libai.so
*    0x0000003777200000  libc.so.6
*    0x00007f6b96657000  libstdc++.so.6
*    0x00007f6b4c931000  driver_aton.so
*    0x0000003777600000  libpthread.so.0
*
* memory: VM 1691 MB, RSS 292 MB, 0 page faults
****
~~~~

Better handling of these unsupported pixel types is needed to prevent this SIGABRT and to inform the user of this limitation. At the very least, the default of the pixel_type switch statement in driver_write_bucket should end the render gracefully with an informative error message.

**Comment by sosoyan** [comment]
Hi Ryan,

I've just tried an ID AOV which is INT, but haven't got any SIGABRT. Do you have Maya or C4D scene so I can double check?

Thanks.

![screen shot 2017-03-14 at 12 08 10](https://cloud.githubusercontent.com/assets/5727306/23891337/3c759dba-08af-11e7-8d1e-caf80e77ae28.png)

**Comment by sosoyan** [comment]
Fixed with 81049c51d0afefc510689b731282f852f7d0c5ae

**Comment by rheniser** [comment]
Thank you for the quick response on this. Sorry for the delay in my response. I have been busy patching AtoN into HtoA (see https://www.linkedin.com/pulse/patching-aton-htoa-ryan-heniser). I will give this fix a test this week. Cheers!
